### PR TITLE
Clean up gossip tests

### PR DESCRIPTION
**Pull request by rustyrussell**
When our use of gossip queries gets more sophisticated (coming PR), the current way we test probing breaks badly.  Use an external helper do ask for gossip instead, which also allows our tests to be more specific and clearer.

### CODE REVIEW

File: tests/test_gossip.py [hunk 1]
```diff
@@ -567,110 +570,167 @@ def test_gossip_query_channel_range(node_factory, bitcoind):
...
-    assert ret['short_channel_ids'][1] == scid23
+    # Asks l2 for all channels, gets both.
+    msgs = l2.query_gossip('query_channel_range',
+                           '06226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910f',
+                           0, 1000000)
+    encoded = subprocess.run(['devtools/mkencoded', '--scids', '00', scid12, scid23],
+                             check=True,
+                             timeout=TIMEOUT,
+                             stdout=subprocess.PIPE).stdout.strip().decode()
+    # reply_channel_range == 264
```
Comment (darosior) [inline tests/test_gossip.py]: 265 ? https://github.com/lightningnetwork/lightning-rfc/blob/master/07-routing-gossip.md#the-query_channel_range-and-reply_channel_range-messages

File: gossipd/queries.c [hunk 2]
```diff
@@ -957,6 +955,45 @@ void maybe_send_query_responses(struct peer *peer)
 	}
 }
 
+bool query_channel_range(struct daemon *daemon,
+			 struct peer *peer,
```
Comment (darosior) [inline gossipd/queries.c]: weird indent ?

File: gossipd/queries.h [hunk 3]
```diff
@@ -22,14 +22,24 @@ void query_unknown_channel(struct daemon *daemon,
 /* This called when the peer is idle. */
 void maybe_send_query_responses(struct peer *peer);
 
+/* Ask this peer for a range of scids.  Must support it, and not already
+ * have a query pending. */
+bool query_channel_range(struct daemon *daemon,
+			 struct peer *peer,
```
Comment (darosior) [inline gossipd/queries.h]: Here too ? Maybe Github rendering..

File: tests/utils.py [hunk 4]
```diff
@@ -750,6 +752,29 @@ def wait_for_onchaind_broadcast(self, name, resolve=None):
...
+                              '{}@localhost:{}'.format(self.info['id'],
+                                                       self.port),
+                              query],
+                             check=True,
+                             timeout=TIMEOUT, stdout=subprocess.PIPE).stdout
+
+        msgs = []
+        while len(out):
+            l = struct.unpack('>H', out[0:2])[0]
+            msgs.append(out[2:2+l].hex())
```
Comment (darosior) [inline tests/utils.py]: `out[2:2 + l].hex()` to make `check-source` happy.

File: tests/test_gossip.py [hunk 5]
```diff
@@ -567,110 +570,167 @@ def test_gossip_query_channel_range(node_factory, bitcoind):
...
+    start = 0
+    scids = '00'
+    for m in msgs:
+        assert m.startswith('0108'
+                            # blockhash
+                            '06226e46111a0b59caaf126043eb5bbf28c34f3a5e332a1fc7b2b73cf188910f')
+        this_start = int(m[4+64:4+64+8], base=16)
+        num = int(m[4+64+8:4+64+8+8], base=16)
+        # Pull off end of packet, assume it's uncompressed, and no TLVs!
+        scids += m[4+64+8+8+2+4+2:]
```
Comment (darosior) [inline tests/test_gossip.py]: linter is dying ^^
```
tests/test_gossip.py:712:29: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:712:34: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:712:37: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:713:22: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:713:25: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:713:29: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:713:32: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:713:34: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:715:21: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:715:24: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:715:26: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:715:28: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:715:30: E226 missing whitespace around arithmetic operator
tests/test_gossip.py:715:32: E226 missing whitespace around arithmetic operator
```

File: gossipd/queries.c (hunk 2 above)
Comment (rustyrussell) [reply gossipd/queries.c]: GH doesn't like tabs it seems...

Comment (ZmnSCPxj) [reply gossipd/queries.c]: It has to do with the `+ ` before the indent.  If you print the `diff` out in your terminal, you will see this phenomenon too, not just on Github.  You can ignore it.

This has to do with the fact that `tab` is not 8 spaces, rather it aligns you to a column such that `((column - 1) % 8) == 0`.  Any characters before a tab will not move the position of where the tab terminates.

Comment (darosior) [reply gossipd/queries.c]: Fwiw the trick I use is to indent with tabs to the precedent line indentation level, then use space to align : git diff, github diff, and my editor are happy.

**Comment by rustyrussell** [comment]
needs rebase on https://github.com/ElementsProject/lightning/pull/3103 to make tests less flaky...

**Comment by rustyrussell** [comment]
Returned failing test to developer-only, since gossip propagation otherwise is too slow and test times out.

**Pull request merged**